services/chat/chatService.ts

```diff
@@ -0,0 +1,34 @@
+'use client'
+
+export type ChatServiceSendMessageParams = {
+    anon_user_info_color_code: string;
+    anon_user_info_color_name: string;
+    anon_user_info_emoji_code: string;
+    anon_user_info_emoji_name: string;
+    member: string;
+    message: string
+}
+
+export type BetterSocialResponse<T> = {
+  success: boolean;
+  message: string;
+  data: T
+}
+
+export type SendMessageResponse = {
+  channel_id: string;
+}
+
+const sendMessage = async (data: ChatServiceSendMessageParams): Promise<BetterSocialResponse<SendMessageResponse>> => {
+  const response = await fetch('/api/chat/send', {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify(data)
+  })
+
+  return await response.json()
+}
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

While the `sendMessage` function is generally well implemented, it lacks error handling. If the fetch request fails due to network issues or if the server returns a non-200 status code, this function will throw an unhandled promise rejection. It's recommended to add a try-catch block to handle potential errors.

```diff
const sendMessage = async (data: ChatServiceSendMessageParams): Promise<BetterSocialResponse<SendMessageResponse>> => {
+  try {
    const response = await fetch('/api/chat/send', {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify(data)
    })

+    if (!response.ok) {
+      throw new Error(`HTTP error! status: ${response.status}`);
+    }

    return await response.json()
+  } catch (error) {
+    console.error('An error occurred while sending the message:', error);
+    return { success: false, message: error.message, data: null };
+  }
}
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->